packages/connector/src/methods.ts

```diff
@@ -16,13 +16,16 @@ import type {
 import { RPCMethods } from './types';
 
 const sendSnapMethod = async <T>(
-  request: unknown,
+  request: { method: string; params?: any },
   snapId: string,
 ): Promise<T> => {
   return await window.ethereum.request({
     method: 'wallet_invokeSnap',
     params: {
-      request,
+      request: {
+        method: request.method,
+        ...(request.params && { params: request.params }),
```

Comment:
```ts
...(request.params && { params: request.params }),
```

What does it actually do? Adds a `params` only if it exists in the `request` object?

Reply:
Very hard to read. I'd rather add an explicit `if`